tests/queries/0_stateless/03362_filter_transform_profile_events.sql

```diff
@@ -0,0 +1,2 @@
+SELECT * FROM system.numbers WHERE number % 2 = 0 LIMIT 100;
+SELECT count() = 2 AS assert_exists FROM system.events WHERE name IN ('FilterTransformPassedRows', 'FilterTransformPassedBytes') HAVING assert_exists;
```

Comment:
I think a better check would be to test the value of the mentioned profile events. You can use the `ProfileEvents` of the `system.query_log` system table ([docs](https://clickhouse.com/docs/operations/system-tables/query_log)). If you search for `ProfileEvents` or `system.query_log` in the tests then you will find nice examples how to use it.

Reply:
Added another test to look at the values of profile events.

Reply:
@antaljanosbenjamin I added the following test but it does not generate deterministic results. As seen in my commit. So, I will remove it. I looked into other ways but there is no way to make sure that same values are returned for `FilterTransformPassedBytes` and `FilterTransformPassedRows` every time, thus making this test flaky.

```
SYSTEM FLUSH LOGS;
SET log_queries = 1;
SELECT ProfileEvents['FilterTransformPassedRows'] AS passed_rows, ProfileEvents['FilterTransformPassedBytes'] AS passed_bytes FROM system.query_log WHERE type = 'QueryFinish' AND current_database = currentDatabase() ORDER BY event_time DESC LIMIT 1;
```

Reply:
I would assume at least `FilterTransformPassedRows` should be deterministic. What causes that not to be? For bytes, I also don't see what cause the flakyness. Maybe I am not seeing everything though, so there might be a good explanation.

Reply:
The flakiness is caused because `system.query_log`'s flushing happens asynchronously. Using `SYSTEM FLUSH LOGS` does not guarantee that latest query stats will be immediately available. This is why I would like to keep the existing test as is. Additionally, other queries might have finished processing and `FilterTransformPassedRows`, `FilterTransformPassedBytes` values reflect their values in the result making it inconsistent.

Reply:
@antaljanosbenjamin thank you for this suggestion. This is working now. I have updated the diff with the extra test. Can we please merge this PR?

Reply:
Hello @antaljanosbenjamin, can this PR be merged?